### ISSUE TITLE
Investigate if Davenport's q-method could be used in the pile-up

### ISSUE THREAD

**Issue by pleroy** [mockingbirdnest/Principia#2966]
This would avoid the cheesy hacky selection of the "reference part".

See [this post](https://math.stackexchange.com/questions/1634113/davenports-q-method-finding-an-orientation-matching-a-set-of-point-samples) and the articles that it references for details.

**Comment by pleroy** [comment]
**TL;DR: Davenport's Q method is more principled and makes the code much simpler than the old heuristic.  It doesn't, however, result in significantly different behaviours, except perhaps in situations involving high aerodynamic forces, which are anyway hard to analyze because they tend to be chaotic.**

The curves below give the angular frequency of the pile-up over time.  The blue curve is for Davenport's Q method, the yellow curve for the old heuristic.

### First test case: the [sounding rocket](https://github.com/mockingbirdnest/Principia/issues/2519#issuecomment-628810018) in #2519:
![rocket](https://user-images.githubusercontent.com/7423020/164542334-fe68e7d0-aaaa-418b-a01c-a59dab1b2f3f.png)
Clearly, there are no visible differences, other than Davenport keeping the rocket intact for a few more seconds.  This vessel is very rigid, so that's not unexpected.

### Second test case: the [extreme spin-up](https://github.com/mockingbirdnest/Principia/issues/2519#issuecomment-629881597) reported by @scimas in #2519.
Before the rapid unscheduled disassembly:
![scimas_stock](https://user-images.githubusercontent.com/7423020/164543067-0ccf8034-e776-4720-9974-f4475f6ff70d.png)
and after:
![scimas_stock2](https://user-images.githubusercontent.com/7423020/164543142-eaebfa00-b56e-44d2-bbb0-8419f7590d59.png)
The time shift in the second graph is believed to be an effect of the chaotic nature of the RUD.  Because the vessel is less rigid that in the previous test case, the curves differ somewhat, but they remain quite similar.  In particular, the absurdly fast spin-up has no physical reality (it should amortize) but it is probably due to KSP being insane and not to the way we handle rotations.

### Third test case: same as above, but with FAR.
The two graphs below show what happens *after* the RUD (the evolution before the RUD is not all that interesting) in two experiments:
![scimas_far](https://user-images.githubusercontent.com/7423020/164544081-28802e6d-bbc4-4cf0-8ac0-6c52216a4e62.png)
![scimas_far2](https://user-images.githubusercontent.com/7423020/164544098-c2deee3d-c368-4745-a9bc-b95ca6901e26.png)
Note first that the chaotic nature of the system makes it really hard to obtain reproducible results.  There are however significant differences in the angular frequency profiles, with Davenport exhibiting less oscillations and slower rotation overall.   Interestingly, FAR results in a much slower spin-up than the stock game, and the oscillations do amortize.  This supports our suspicion that the stock aerodynamic has serious problems.

**Comment by JovietUnion** [comment]
Looks like ventricular tachycardia, tbh
